packages/react-components/react-breadcrumb/src/utils/partitionBreadcrumbItems.test.ts

```diff
@@ -25,9 +25,21 @@ const testData = [
     { items, maxDisplayedItems: 9, overflowIndex: 9 },
     { startDisplayedItems: [0, 1, 2, 3, 4, 5, 6, 7], overflowItems: [8, 9], endDisplayedItems: [10] },
   ],
+  [
```

Comment:
Looks like those type errors in build might be fixed if we introduce a type for the test data.
`type TestData = [PartitionBreadcrumbItemsOptions<number>, ReturnType<typeof partitionBreadcrumbItems>][];`
And to use it for `testData` and `maxDisplayedItemsData`, e.g.:
`const maxDisplayedItemsData: TestData = [ ... ];`

Reply:
@GordonSmith , if you want I can commit those changes

Reply:
Sure - I am travelling ATM, but when I get a chance I will ammend with your code...

Note:  I am not 100% convinced that this PR is correct (with what I instinctively expected) - for example in some tests there is always an endDisplayItems array, even when I didn't expect it.  But I do think it is a closer match to current behaviour with some better out of bounds handling.